app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+
+        expect(blockTracker.isRunning()).toBe(false);
       });
     });
+  });
 
-    describe('#setProviderType', () => {
-      it('should update provider.type', async () => {
-        await withController(async ({ controller }) => {
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.RPC });
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.MAINNET });
-          await controller.initializeProvider();
-          controller.setProviderType('mainnet');
-          const { type } = controller.getProviderConfig();
-          expect(type).toStrictEqual('mainnet');
+  describe('initializeProvider', () => {
+    it('throws if the provider configuration is invalid', async () => {
+      const invalidProviderConfig = {};
+      await withController(
+        {
+          state: {
+            provider: invalidProviderConfig,
+          },
+        },
+        async ({ controller }) => {
+          await expect(async () => {
+            await controller.initializeProvider();
+          }).rejects.toThrow(
+            'NetworkController - _configureProvider - unknown type "undefined"',
+          );
+        },
+      );
+    });
+
+    for (const {
+      networkName,
+      networkType,
+      chainId,
+      networkVersion,
+    } of INFURA_NETWORKS) {
+      describe(`when the type in the provider configuration is "${networkType}"`, () => {
+        it(`initializes a provider pointed to the ${networkName} Infura network (chainId: ${chainId})`, async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: networkType,
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls();
+
+              await controller.initializeProvider();
+
+              const { provider } = controller.getProviderAndBlockTracker();
+              const promisifiedSendAsync = promisify(provider.sendAsync).bind(
+                provider,
+              );
+              const chainIdResult = await promisifiedSendAsync({
+                method: 'eth_chainId',
+              });
+              expect(chainIdResult.result).toBe(chainId);
+            },
+          );
         });
-      });
 
-      it('should set the network to loading', async () => {
-        await withController(async ({ controller }) => {
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.RPC });
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.MAINNET });
-          await controller.initializeProvider();
+        it('emits infuraIsUnblocked (assuming that the request to eth_blockNumber responds successfully)', async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: 'rpc',
+                  rpcUrl: 'https://mock-rpc-url',
+                  chainId: '0x1337',
+                },
```

Comment:
I'm guessing this was meant to be set to the network type for this describe block:
```suggestion
                provider: {
                  type: networkType,
                },
```
Easy to miss because the test passes either way